client_cli/pytests/src/client_cli/client_cli.py

```diff
@@ -0,0 +1,255 @@
+    A class to represent the Iroha client command line interface.
+    """
+
+    def __init__(self, config: Config, path: str):
+        """
+        Constructs all the necessary attributes for the ClientCli object.
+
+        :param config: The configuration object.
+        :type config: Config
+        :param path: Path to the Iroha client command line interface executable.
+        :type path: str
+        """
+        self.config = config
+        self.default_command = [path, f'--config={self.config.file}']
+        self.command = self.default_command.copy()
+        self.stdout = None
+        self.stderr = None
+        self._timeout = 5
+
+    def __enter__(self):
+        """
+        Called when entering a context managed by the ClientCli object.
+        """
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        Called when exiting a context managed by the ClientCli object.
+
+        :param exc_type: The type of exception raised within the context (if any).
+        :param exc_val: The instance of the exception raised within the context (if any).
+        :param exc_tb: A traceback object encapsulating the call stack at the point
+                        where the exception was raised (if any).
+        """
+        self.reset()
+
+    def wait_for(self, expected: str, actual: str, method: Callable, timeout=None):
+        """
+        Wait for a certain condition to be met, specified by the expected and actual values.
+
+        :param expected: The expected value.
+        :type expected: str
+        :param actual: The actual value.
+        :type actual: str
+        :param method: A callable method to evaluate the condition.
+        :type method: Callable
+        :param timeout: Maximum time to wait for the condition to be met, defaults to None.
+        :type timeout: int, optional
+        """
+        timeout = timeout or self._timeout
+        start_time = time()
+        while expected not in actual:
+            if time() - start_time > timeout:
+                allure.attach(actual, name='actual', attachment_type=allure.attachment_type.TEXT)
+                allure.attach(expected, name='expected', attachment_type=allure.attachment_type.TEXT)
+                raise TimeoutError(f"Expected '{expected}' "
+                                   f"to be in '{actual}' "
+                                   f"after waiting for '{timeout}' seconds.")
+            sleep(1)
+            actual = method()
+        assert expected in actual
+
+    def reset(self):
+        """
+        Resets the stdout and stderr attributes of the ClientCli object.
+        """
+        self.stdout = None
+        self.stderr = None
+
+    def register(self):
+        """
+        Appends the 'register' command to the command list.
+
+        :return: The current ClientCli object.
+        :rtype: ClientCli
+        """
+        self.command.append('register')
+        return self
+
+    def mint(self):
```

Comment:
you can also add `burn`. It's been implemented in the `iroha_client_cli`

Reply:
Created #3566